### PR TITLE
[Merged by Bors] - feat: port Analysis.SpecialFunctions.CompareExp

### PR DESCRIPTION
**Pull request by apurvnakade**
I had to add several `norm_cast` to force coercion. My guess is that in Lean3 `simp` also tried `norm_cast` but here we need to invoke the tactic explicitly.

There are two `calc` proofs that I'm having difficulty resolving, both because of type casting issues.

- [x] isLittleO_im_pow_exp_re
- [x] isLittleO_cpow_exp

Please feel free to push to this branch.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/CompareExp.lean [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+import Mathlib.Analysis.Asymptotics.AsymptoticEquivalent
+import Mathlib.Analysis.Asymptotics.SpecificAsymptotics
+
+/-!
+# Growth estimates on `x ^ y` for complex `x`, `y`
+
+Let `l` be a filter on `ℂ` such that `Complex.re` tends to infinity along `l` and `Complex.im z`
+grows at a subexponential rate compared to `Complex.re z`. Then
+
+- `Complex.is_o_log_abs_re_of_subexponential_im_re`: `Real.log ∘ Complex.abs` is `o`-small of
```
Comment (apurvnakade) [inline Mathlib/Analysis/SpecialFunctions/CompareExp.lean:21]: This docstring is incorrect in Mathlib3. There is no function `is_o_log_abs_re_of_subexponential_im_re`. 

What to do here?

**Comment by apurvnakade** [comment]
I'm completely stuck at these `calc` proofs and not able to make any progress.

File: Mathlib/Analysis/SpecialFunctions/CompareExp.lean [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  Real.isLittleO_log_id_atTop.comp_tendsto hl.tendsto_re
+#align complex.is_exp_cmp_filter.is_o_log_re_re Complex.IsExpCmpFilter.isLittleO_log_re_re
+set_option maxHeartbeats 0
+theorem isLittleO_im_pow_exp_re (hl : IsExpCmpFilter l) (n : ℕ) :
+    (fun z : ℂ => z.im ^ n) =o[l] fun z => Real.exp z.re :=
+  flip IsLittleO.of_pow two_ne_zero <|
+    calc
+      ((fun z : ℂ => (z.im ^ n)) ^ 2) = (fun z : ℂ => (z.im ^ n) ^ 2) := funext <| by simp
+      _ = fun z => (Complex.im z) ^ (2 * n) := funext <| fun _ => by norm_cast ; rw [pow_mul']
+      _ =O[l] fun z => Real.exp z.re := sorry --(hl.isBigO_im_pow_re _)
```
Comment (apurvnakade) [inline Mathlib/Analysis/SpecialFunctions/CompareExp.lean:121]: My Lean server crashes if I uncomment this.

File: Mathlib/Analysis/SpecialFunctions/CompareExp.lean (hunk 1 above)
Comment (int-y1) [reply Mathlib/Analysis/SpecialFunctions/CompareExp.lean:21]: You can change it here. Doc fixes are safe to do.
Fixing the mathlib3 docs is optional.

**Comment by int-y1** [comment]
`isLittleO_cpow_exp`: I added `show ... from` in order to cast the function from `(z : ℂ) → (fun x ↦ ℝ) z` to `ℂ → ℝ`

File: Mathlib/Analysis/SpecialFunctions/CompareExp.lean (hunk 1 above)
Comment (apurvnakade) [reply Mathlib/Analysis/SpecialFunctions/CompareExp.lean:21]: Thanks!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**